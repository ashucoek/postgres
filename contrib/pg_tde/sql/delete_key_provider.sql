CREATE EXTENSION IF NOT EXISTS pg_tde;

SELECT  * FROM pg_tde_key_info();

SELECT pg_tde_add_database_key_provider_file('file-provider','/tmp/pg_tde_test_keyring.per');
SELECT * FROM pg_tde_list_all_database_key_providers();
SELECT pg_tde_delete_database_key_provider('file-provider');
SELECT * FROM pg_tde_list_all_database_key_providers();

SELECT pg_tde_add_database_key_provider_file('file-provider','/tmp/pg_tde_test_keyring.per');
SELECT * FROM pg_tde_list_all_database_key_providers();
SELECT pg_tde_delete_database_key_provider('file-provider');
SELECT * FROM pg_tde_list_all_database_key_providers();

SELECT pg_tde_add_database_key_provider_file('file-provider','/tmp/pg_tde_test_keyring.per');
SELECT * FROM pg_tde_list_all_database_key_providers();
SELECT pg_tde_delete_database_key_provider('file-provider');
SELECT * FROM pg_tde_list_all_database_key_providers();

DROP EXTENSION pg_tde;
