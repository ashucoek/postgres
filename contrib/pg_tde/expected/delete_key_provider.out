CREATE EXTENSION IF NOT EXISTS pg_tde;
SELECT  * FROM pg_tde_key_info();
ERROR:  Principal key does not exists for the database
HINT:  Use set_key interface to set the principal key
SELECT pg_tde_add_database_key_provider_file('file-provider','/tmp/pg_tde_test_keyring.per');
 pg_tde_add_database_key_provider_file 
---------------------------------------
                                     1
(1 row)

SELECT * FROM pg_tde_list_all_database_key_providers();
 id | provider_name | provider_type |                          options                           
----+---------------+---------------+------------------------------------------------------------
  1 | file-provider | file          | {"type" : "file", "path" : "/tmp/pg_tde_test_keyring.per"}
(1 row)

SELECT pg_tde_delete_database_key_provider('file-provider');
 pg_tde_delete_database_key_provider 
-------------------------------------
 
(1 row)

SELECT * FROM pg_tde_list_all_database_key_providers();
 id | provider_name | provider_type | options 
----+---------------+---------------+---------
(0 rows)

SELECT pg_tde_add_database_key_provider_file('file-provider','/tmp/pg_tde_test_keyring.per');
 pg_tde_add_database_key_provider_file 
---------------------------------------
                                     2
(1 row)

SELECT * FROM pg_tde_list_all_database_key_providers();
 id | provider_name | provider_type |                          options                           
----+---------------+---------------+------------------------------------------------------------
  2 | file-provider | file          | {"type" : "file", "path" : "/tmp/pg_tde_test_keyring.per"}
(1 row)

SELECT pg_tde_delete_database_key_provider('file-provider');
 pg_tde_delete_database_key_provider 
-------------------------------------
 
(1 row)

SELECT * FROM pg_tde_list_all_database_key_providers();
 id | provider_name | provider_type | options 
----+---------------+---------------+---------
(0 rows)

SELECT pg_tde_add_database_key_provider_file('file-provider','/tmp/pg_tde_test_keyring.per');
 pg_tde_add_database_key_provider_file 
---------------------------------------
                                     3
(1 row)

SELECT * FROM pg_tde_list_all_database_key_providers();
 id | provider_name | provider_type |                          options                           
----+---------------+---------------+------------------------------------------------------------
  3 | file-provider | file          | {"type" : "file", "path" : "/tmp/pg_tde_test_keyring.per"}
(1 row)

SELECT pg_tde_delete_database_key_provider('file-provider');
 pg_tde_delete_database_key_provider 
-------------------------------------
 
(1 row)

SELECT * FROM pg_tde_list_all_database_key_providers();
 id | provider_name | provider_type | options 
----+---------------+---------------+---------
(0 rows)

DROP EXTENSION pg_tde;
